src/stats.h

```diff
@@ -0,0 +1,72 @@
+#ifndef _STATS_LIB_
+#define _STATS_LIB_
+
+#include "linear_congruential_random_generator.h"
+#include <limits.h>
+#include <math.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifndef EPISLON
+#define EPSILON 0.01
+#endif
+
+#ifndef REAL_NUMBER
+#define REAL_NUMBER float
+#endif
+
+#ifndef MAX_REAL_NUMBER
+#define MAX_REAL_NUMBER __FLT_MAX__
+#endif
+
+#ifndef MIN_REAL_NUMBER
+#define MIN_REAL_NUMBER (-(MAX_REAL_NUMBER)-1)
+#endif
+
+#ifndef INTEGER
+#define INTEGER unsigned int
+#endif
+
+typedef REAL_NUMBER real_number;
+typedef INTEGER vec_size;
```

Comment:
Why are we using vec_size here if we are not working with vectors ?

Reply:
vec_size refers to a vector as a sequence of elements which applies in this case.